### PR TITLE
TravisCI integration

### PR DESCRIPTION
**Pull request by realityforge**
TravisCI integration driven by the google/elemental2#82 issue.

This has yet to be successfully tested as master does not currently build. A fix will be submitted in a separate pull request.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  - echo "startup --host_jvm_args=-Xms2500m" >> .bazelrc
+  # Avoid progress indicators that can blow out log which makes using WEB UI difficult
+  - echo "common --noshow_progress" >> .bazelrc
+  - echo "common --noshow_loading_progress" >> .bazelrc
+  # This is so we understand failures better
+  - echo "build --verbose_failures" >> .bazelrc
+
+script:
+  - bazel build //java/jsinterop/base:libbase.jar
+  - bazel build //java/jsinterop/base:libbase-src.jar
```
Comment (jDramaix) [inline .travis.yml]: bazel build java/...

**Comment by jDramaix** [comment]
The fix for building jsinterop-base on mac is discussed on #8 

I'll do the fix internally and then push here

**Comment by realityforge** [comment]
Force pushed another update to move to using ./bazel_build_test.sh and bazel 0.23.0

**Comment by gkdn** [comment]
Is there any way to reuse travis configuration across projects?

**Comment by realityforge** [comment]
No that I am aware of :(

Internally we generate the `.travis.yml` from a project descriptor but that relies on in-house tooling and is no where near general enough.

**Comment by gkdn** [comment]
Our internal CI that we use for open-source already has can publish build status and add the badge. Is it worthwhile to do Travis?

**Comment by realityforge** [comment]
I am not sure.

For the google internal team - I suspect it offers little to no value.

For those of us outside it is nice for the following reasons:

* TravisCI can be setup to run on pull requests. This gives a green status when succeeded and red status and log when error occurs.
* TravisCI can run on forks (I tend to push changes to a branch in my local fork and make sure it builds)
* It gives us a very clear way of tracking down environmental errors. At works case you can use docker and Travis images to run builds locally and see what is happening. 
* Build logs!
* It builds against OSX and Linux - whereas locally I only use OSX but it is good to verify it works on other OSes. It is unclear how much work it would be to get it building against TravisCI windows images but that could be an aspirational goal ;)

I would also like to see mail outs to some central list so people can get push notifications when a repository starts to fail. Although that is down the track somewhat.

To minimize some of the cruft in the `.travis.yml` configuration it was suggested that I investigate [bazelisk](https://github.com/philwo/bazelisk) which could help manage the bazel version and install. Essentially you store the version in `.bazelversion` and run bazelisk and it is responsible for doing the magic of making sure bazel is up to date. I have yet to investigate to see if it actually improves the situation but it was an option.

**Comment by gkdn** [comment]
Ok these are all fair arguments. It just unfortunate to update 4 copies of the same file; if we could reduce the duplication that would help a lot.

**Comment by realityforge** [comment]
Thanks!

**Pull request closed without merging**